### PR TITLE
test: [M3-6052] - Add LKE cluster update and delete integration tests

### PR DESCRIPTION
**Pull request by jdamore-linode**
## Description 📝
Adds a handful of integration tests to verify Cloud Manager UI behaves as expected when interacting with LKE clusters. 

Opted to use mocked API data for the sake of performance since there's a lot going on that would otherwise require some pretty time consuming set-up and teardown.

## Major Changes 🔄
**List highlighting major changes**
- LKE delete flows confirm that
  - ...LKE clusters can be deleted from the LKE landing page and from cluster detail pages
  - ...UI updates to reflect deleted clusters
- LKE update flows confirm that
  - ...clusters can be upgraded to high availability control plane
  - ...users are notified when their cluster is not on the latest version of Kubernetes
  - ...clusters can be upgraded to the latest version of Kubernetes
  - ...nodes, node pools, and entire clusters can be recycled
  - ...autoscaling can be enabled and disabled
  - ...pools can be resized
  - ...pools can be added and deleted
- LKE landing page tests confirm that
  - ...clusters are listed on the landing page
  - ...a welcome page is shown when there are no clusters
  - ...kubeconfig files can be downloaded from the landing page
- Ventured into `src/` to make the following changes:
  - Added a couple QA data attributes to make working with node pools easier
  - Added a Kubernetes dashboard URL response factory
  - Quick fix to dismiss the node recycle dialog upon confirmation

## How to test 🧪
Run `yarn && yarn build && yarn start:manager:ci`, and then run:

```bash
yarn cy:run -s "cypress/e2e/kubernetes/lke-delete.spec.ts,cypress/e2e/kubernetes/lke-update.spec.ts,cypress/e2e/kubernetes/lke-landing-page.spec.ts"
```

Confirm that the tests pass.

Alternatively, to run the tests interactively so you can view the interactions, run `yarn cy:debug` then search for 'lke' to find relevant spec files.

### CODE REVIEW

**Comment by jaalah-akamai** [comment]
Screen Shot 2023-04-25 at 10 26 19 PM

Looks good - thanks @jdamore-linode

**Review by jaalah-akamai — APPROVED** [review]

**Review by bnussman-akamai — APPROVED** [review]
Awesome 🚀

File: packages/manager/cypress/e2e/kubernetes/lke-update.spec.ts [hunk 1]
```diff
@@ -0,0 +1,744 @@
...
+          .findByTitle('Recycle')
+          .should('be.visible')
+          .should('be.enabled')
+          .click();
+      });
+
+    cy.wait('@recycleNode');
+    ui.toast.assertMessage('Node queued for recycling.');
+
+    // For now, we have to manually cancel out of this dialog.
```
[the remark this answers is not in the document]
Comment (jdamore-linode) [reply packages/manager/cypress/e2e/kubernetes/lke-update.spec.ts]: Thanks! Just pushed that

**Review by cpathipa — APPROVED** [review]
LGTM!

**Pull request merged**